pkg/kv/kvserver/batcheval/cmd_add_sstable.go

```diff
@@ -69,7 +78,9 @@ func EvalAddSSTable(
 	// observed or closed.
 	if args.WriteAtRequestTimestamp &&
 		(args.SSTTimestamp.IsEmpty() || h.Timestamp != args.SSTTimestamp) {
-		sst, err = storage.UpdateSSTTimestamps(sst, h.Timestamp)
+		// TODO(dt): use a quotapool.
+		concurrency := int(addSSTableRewriteConcurrency.Get(&cArgs.EvalCtx.ClusterSettings().SV))
+		sst, err = storage.UpdateSSTTimestamps(sst, args.SSTTimestamp, h.Timestamp, concurrency)
```

Comment:
This is going to fail for clients that don't set `SSTTimestamp`, but that's ok for now since it's not long for this world anyway.

Reply:
Is it? I thought the IsEmpty condition would handle that?

Reply:
You're right, missed that.